### PR TITLE
Additional serialization classes movement

### PR DESCRIPTION
**Pull request by pveentjer**
Moved to com.hazelcast.internal.serialization:
BufferedObjectDataInput
BufferedObjectDataOutput
UTFEncoderDecoder
The unit tests for the com.hazelcast.internal.serialization package

### CODE REVIEW

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/DataInputOutputTest.java [hunk 1]
```diff
@@ -43,7 +39,7 @@
 @Category(QuickTest.class)
 public class DataInputOutputTest {
 
-    private final Person person = new Person(111, 123L, 89.56d, "test-person", new Address("street", 987));
+    private final SerializationConcurrencyTest.Person person = new SerializationConcurrencyTest.Person(111, 123L, 89.56d, "test-person", new SerializationConcurrencyTest.Address("street", 987));
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/DataInputOutputTest.java]: You lost the static imports of `Address` and `Person` here (seems to be an issue with IntelliJ and moving files).

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/PacketTest.java [hunk 2]
```diff
@@ -41,24 +36,24 @@
 @Category(QuickTest.class)
 public class PacketTest {
 
-    private final Person person = new Person(111, 123L, 89.56d, "test-person", new Address("street", 987));
+    private final SerializationConcurrencyTest.Person person = new SerializationConcurrencyTest.Person(111, 123L, 89.56d, "test-person", new SerializationConcurrencyTest.Address("street", 987));
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/PacketTest.java]: Same here:
```
-import static com.hazelcast.nio.serialization.SerializationConcurrencyTest.Address;		
-import static com.hazelcast.nio.serialization.SerializationConcurrencyTest.FACTORY_ID;		
-import static com.hazelcast.nio.serialization.SerializationConcurrencyTest.Person;		
-import static com.hazelcast.nio.serialization.SerializationConcurrencyTest.PortableAddress;		
-import static com.hazelcast.nio.serialization.SerializationConcurrencyTest.PortablePerson;
```

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/PortableClassVersionTest.java [hunk 3]
```diff
@@ -157,10 +158,10 @@ static void testDifferentClassVersionsUsingDataWriteAndRead(SerializationService
 
     @Test
     public void testPreDefinedDifferentVersionsWithInnerPortable() {
-        final SerializationService serializationService = createSerializationService(1);
+        final SerializationService serializationService = PortableTest.createSerializationService(1);
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/PortableClassVersionTest.java]: The static import is there but unused (4 times), also lost import of `createNamedPortableClassDefinition`.

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/PortableTest.java [hunk 4]
```diff
@@ -337,7 +348,7 @@ public void testClassDefinition_getNestedField() throws IOException {
 
         FieldDefinition fd = portableContext.getFieldDefinition(classDefinition, "child");
         assertNotNull(fd);
-        assertEquals(FieldType.PORTABLE, fd.getType());
+        Assert.assertEquals(FieldType.PORTABLE, fd.getType());
```
Comment (Donnerbart) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/PortableTest.java]: Static import of assertEquals() got lost.

**Comment by Donnerbart** [comment]
Just minor import issues, which is an issue I encountered myself when moving files with IntelliJ. Should be easy to fix.

The movement of the files itself seems to be okay, also the suppressions.xml looks fine.

Under the premise that those files are all private API: :+1:

**Comment by asimarslan** [comment]
:+1:

File: hazelcast/src/test/java/com/hazelcast/internal/serialization/DataInputOutputTest.java (hunk 1 above)
Comment (pveentjer) [inline hazelcast/src/test/java/com/hazelcast/internal/serialization/DataInputOutputTest.java]: I'll fix them!

**Comment by pveentjer** [comment]
Resolved the review comments. Thanks!

**Comment by sancar** [comment]
:+1:

**Comment by sancar** [comment]
@pveentjer It seems this pr needs your attention.

**Comment by sancar** [comment]
@pveentjer What are we gonna do with this one ?

**Comment by enesakar** [comment]
@pveentjer I moved this to 3.7. If these refactoring is not needed after new protocol; you can close it.

**Pull request closed without merging**